apps/src/netsim/NetSimHeartbeat.js

```diff
@@ -133,12 +141,30 @@ NetSimHeartbeat.prototype.setBeatInterval = function (intervalMs) {
 };
 
 /**
+ * Set a handler to call if this heartbeat is unable to update its remote
+ * storage representation.  Can be used to go into a recovery mode,
+ * acknowledge disconnect, and/or attempt an auto-reconnect.
+ * @param {function} onFailedHeartbeat
+ */
+NetSimHeartbeat.prototype.setFailureCallback = function (onFailedHeartbeat) {
+  this.onFailedHeartbeat_ = onFailedHeartbeat;
+};
+
+/**
  * Updates own row on regular interval, as long as something's making
  * it tick.
  */
 NetSimHeartbeat.prototype.tick = function () {
   if (Date.now() - this.time_ > this.intervalMs_) {
     this.time_ = Date.now();
-    this.update();
+    this.update(function (success) {
+      if (!success) {
```

Comment:
In node land, callbacks usually take the form callback(err, result). Might make more sense to emulate that and send an error (or no error) instead of success.